ledger/ledger-api-common/src/main/scala/com/digitalasset/platform/server/api/validation/ErrorFactories.scala

```diff
@@ -476,9 +486,15 @@ class ErrorFactories private (errorCodesVersionSwitcher: ErrorCodesVersionSwitch
     * @param status A Protobuf [[Status]] object.
     * @return An exception without a stack trace.
     */
-  def grpcError(status: Status): StatusRuntimeException = new NoStackTraceApiException(
-    StatusProto.toStatusRuntimeException(status)
-  )
+  def grpcError(status: Status): StatusRuntimeException = {
+    val maxMessageLength = 1536
```

Comment:
This value owes an explanation in the comment.

Reply:
Added a comment detailing why we need to truncate. Re the actual numerical limit justification, I would defer to @daravep (maybe you have an heuristic from Canton where this problem occurs more frequently). The upper limit of what we can use there is 8K (default Netty header size), but that can include other KV pairs.

Reply:
I'd say the following: the message is human centric. Any message beyond 1.5k is not a message anymore, but an encyclopedia. Though the status runtime exception call this field really message. Hence, let's limit it.

I don't have a precise heuristic on how to compute the allowed size. I even think it is not possible, because we are not talking just about the 8k size of the exception. There is other stuff transported as part of the header which we can't influence.

Reply:
Is this length in bytes?